frontend/src/metabase/query_builder/components/CompareAggregations/CompareAggregations.tsx

```diff
@@ -67,7 +72,49 @@ export const CompareAggregations = ({
         columns,
         offset,
       );
-      onSubmit(aggregations);
+
+      if (!columnAndBucket) {
+        return;
+      }
+
+      let nextQuery = aggregations.reduce(
```

Comment:
This feature breaks encapsulation a little bit in that it makes the `CompareAggregations` component edit the query quite extensively: it needs to move or add a breakout. For this, it needs the `updateQuery` function, instead of just being able to pass aggregations via `onAdd`.

Reply:
> For this, it needs the `updateQuery` function, instead of just being able to pass aggregations via `onAdd`.

We shouldn't have 2 different approaches to this. If `onAddAggregations` and `onAddAggregation` no longer suffice, then perhaps we should get rid of them.

How about this signature to replace `onAddAggregations`, `onUpdateAggregation`, and `updateQuery`?

```tsx
onChange(query: Lib.Query, details: {
  aggregations: Lib.ExpressionClause[];
  type: "create" | "update";
});
```

Reply:
Or maybe just call `updateQuery`? This `onChange` looks complex

Reply:
We need the aggregations for some other reasons than just to update the query (ie. to log to analytics), but I guess we can extract them from the query too.

Reply:
They are the aggregations that are getting added or created.

What's more: in some cases we don't just want to know that the query changed, but _how and why_ too. For example `useSummarizeQuery` handles aggregation removals with some special logic.